pkg/daemonset/daemonset.go

```diff
@@ -174,6 +174,28 @@ func (builder *Builder) WithHostNetwork() *Builder {
 	return builder
 }
 
+// WithPodAffinity applies pod's Affinity to daemonset definition.
+func (builder *Builder) WithPodAffinity(podAffinity *corev1.Affinity) *Builder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	if podAffinity == nil {
+		glog.V(100).Infof("The Affinity parameter is empty")
+
+		builder.errorMsg = "Affinity parameter is empty"
```

Comment:
one last minor point but errorMsg should start with a lowercase
```suggestion
		builder.errorMsg = "affinity parameter is empty"
```